### DISCUSSION TITLE
Perferred case question ?!

### DISCUSSION THREAD

**Discussion by syf1913** [krvajal/vscode-fortran-support#738] [Q&A - deprecated see Fortran Discourse]
Hi, : )
`Perferred case` is set to `uppercase`, but when using auto-completion, the complementary code fragment is not in `uppercase`?

![20221102182344](https://user-images.githubusercontent.com/89260210/199465824-f225a198-eea1-4788-8020-0a00d4a444a0.png)

**Comment by gnikit** [comment]
## Temporary fix:

You want to pass false to the flag `--lowercase_intrinsics` to fortls (the language server) not the vscode extension. Given how the command line interface is currently written in fortls that won't be possible but you can create a config file for fortls to override the default arguments.

```json
{
 "lowercase_intrinsics": false
}
```

See docs for more: https://gnikit.github.io/fortls/options.html#configuration-using-a-file

## long term solution

I will fix the vscode settings to not pass --lowercase_intrinsics if you have selected uppercase as your preferred case

**Reply by syf1913** [reply]
ok, I see. Thanks a lot ! : )

**Reply by syf1913** [reply]
Hang on a sec， the default argument of `lowercase_intrinsics` is `false`.
And I creat a `.fortlsrc` file with `"lowercase_intrinsics": false`  under the `root_path` of the project, but It doesn't seem to work. : (

**Reply by gnikit** [reply]
Upon a closer look, you are already getting lowercase intrinsics and autocompletions. What cannot be lowercase is the snippets.
The `do` completion suggestion that you are getting is a hardwired snippet, written in JSON. There might be a way to programmatically write snippets for VS Code extensions but I am not sure if it's worth it for Fortran.

I hope this clears things up a bit.
